apps/api/src/app/subscribers-v2/usecases/create-subscriber/create-subscriber.usecase.ts

```diff
@@ -0,0 +1,36 @@
+import { Injectable } from '@nestjs/common';
+import { SubscriberEntity, SubscriberRepository } from '@novu/dal';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { SubscriberResponseDto } from '../../../subscribers/dtos';
+import { mapSubscriberEntityToDto } from '../list-subscribers/map-subscriber-entity-to.dto';
+import { CreateSubscriberCommand } from './create-subscriber.command';
+
+@Injectable()
+export class CreateSubscriber {
+  constructor(private subscriberRepository: SubscriberRepository) {}
+
+  async execute(command: CreateSubscriberCommand): Promise<SubscriberResponseDto> {
+    const existingSubscriber = await this.subscriberRepository.findOne({
+      subscriberId: command.createSubscriberRequestDto.subscriberId,
+      _environmentId: command.environmentId,
+      _organizationId: command.organizationId,
+    });
+
+    if (existingSubscriber) {
+      throw new ApiException(`Subscriber: ${command.createSubscriberRequestDto.subscriberId} already exists`);
+    }
+
+    const nonUndefinedEntries = Object.entries(command.createSubscriberRequestDto).filter(
```

Comment:
Is this clean up necessary? If not mistaken mongoose has a way to ignore undefined values. See [here](https://mongoosejs.com/docs/guide.html#minimize).

Reply:
Yeah right, it wont be needed here.
removing that part